iosApp/iosApp/ViewModels/StopDetailsViewModel.swift

```diff
@@ -65,10 +67,12 @@ class StopDetailsViewModel: ObservableObject {
     private let tripRepository: ITripRepository
     private let vehicleRepository: IVehicleRepository
 
-    private let togglePinnedUsecase = UsecaseDI().toggledPinnedRouteUsecase
+    private let favoritesUsecases: FavoritesUsecases
+    private let togglePinnedUsecase: TogglePinnedRouteUsecase
```

Comment:
It turns out that if you allow overriding the `Repository` you need to correctly use that repository in the `UseCase` or else you will be completely miserable. I was printing the repository itself as a string (since that’s derived from its memory address) in like five different places trying to figure out where it became the wrong repository before I finally tracked this down.